### PR TITLE
Poshi 219 & Poshi 229 Update validation syntax and begin method migration for IDE

### PR DESCRIPTION
**Pull request by CalumR23**
https://issues.liferay.com/browse/POSHI-229
https://issues.liferay.com/browse/POSHI-219

### CODE REVIEW

**Comment by kenjiheigel** [comment]
Forwarded to BChan.
https://github.com/brianchandotcom/liferay-portal/pull/112287

File: modules/test/poshi/poshi-core/src/main/java/com/liferay/poshi/core/PoshiValidation.java [hunk 1]
```diff
@@ -1016,6 +1017,53 @@ protected static void validateMessageElement(
 		}
 	}
 
+	protected static void validateMethodElement(
+		Element element, String filePath) {
+
+		String attributeName = element.attributeValue("method");
```
Comment (CalumR23) [inline modules/test/poshi/poshi-core/src/main/java/com/liferay/poshi/core/PoshiValidation.java:1023]: attrValue & methodValue

File: modules/test/poshi/poshi-core/src/main/java/com/liferay/poshi/core/PoshiValidation.java [hunk 2]
```diff
@@ -1016,6 +1017,53 @@ protected static void validateMessageElement(
...
+
+		if ((attributeName != null) && attributeName.contains("selenium")) {
+			Matcher matcher = _seleniumPattern.matcher(attributeName);
+
+			while (matcher.find()) {
+				String seleniumMethod = matcher.group(1);
+
+				String methodName = seleniumMethod.substring(
```
Comment (CalumR23) [inline modules/test/poshi/poshi-core/src/main/java/com/liferay/poshi/core/PoshiValidation.java:1031]: make the regex more specific

File: modules/test/poshi/poshi-core/src/main/java/com/liferay/poshi/core/PoshiValidation.java [hunk 3]
```diff
@@ -1016,6 +1017,53 @@ protected static void validateMessageElement(
...
+					seleniumMethod.indexOf("#") + 1);
+
+				if (!methodName.equals("getCurrentUrl")) {
+					String parameters = matcher.group(2);
```
Comment (CalumR23) [inline modules/test/poshi/poshi-core/src/main/java/com/liferay/poshi/core/PoshiValidation.java:1035]: -> parameterNames

File: modules/test/poshi/poshi-core/src/main/java/com/liferay/poshi/core/PoshiValidation.java [hunk 4]
```diff
@@ -1016,6 +1017,53 @@ protected static void validateMessageElement(
...
+					int parameterCount = PoshiContext.getSeleniumParameterCount(
+						methodName);
+
+					List<String> parameterList =
+						PoshiParserUtil.getMethodParameters(parameters);
```
Comment (kenjiheigel) [inline modules/test/poshi/poshi-core/src/main/java/com/liferay/poshi/core/PoshiValidation.java:1040]:
```
String parameters = matcher.group(2);

...

List<String> parameterList = PoshiParserUtil.getMethodParameters(parameters);
```
Should be: 
```
List<String> parameters = PoshiParserUtil.getMethodParameters(matcher.group(2));
```

**Pull request closed without merging**